rdflib/graph.py

```diff
@@ -400,6 +409,17 @@ def addN(self, quads):
             and _assertnode(s, p, o)
         )
 
+    def addNAll(self, quads):
+        """Add a sequence of triples with context"""
+        for i in quads:
+            self.__store.addN(
+                (s, p, o, c)
+                for s, p, o, c in i
```

Comment:
I think this is a bug. Was this method tested at all? `i` is already a quad item from the list of quads, you cannot unpack it futher into more quads using this `for .. in` statement.